### PR TITLE
`array_interpolate_linear`: Apply interpolation to NaN values

### PR DESCRIPTION
**Pull request by m-mohr**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- `array_interpolate_linear`: Apply interpolation also on NaN values.
```
Comment (m-mohr) [inline CHANGELOG.md:11]:
```suggestion
- `array_interpolate_linear`: Apply interpolation to NaN values.
```

**Comment by clausmichele** [comment]
Do we need two types: no-data and NaN ?

**Comment by m-mohr** [comment]
They are pretty much the same in the context of openEO.

**Comment by m-mohr** [comment]
Opened #480

**Comment by m-mohr** [comment]
After the recent discussions, I think this PR is fine as it is. The aim for this process is to fill "holes" in the data and both NaN and any additional no-data values that may apply should be filled.

**Comment by m-mohr** [comment]
Closing in favor of #490

**Pull request closed without merging**